specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json

```diff
@@ -1368,6 +1368,379 @@
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "updateSqlDatabaseParameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SqlDatabaseCreateUpdateParameters"
+            },
+            "description": "The parameters to provide for the current SQL database."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The SQL database update operation was successful.",
+            "schema": {
+              "$ref": "#/definitions/SqlDatabase"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "DatabaseAccounts_DeleteSqlDatabase",
+        "x-ms-examples": {
+          "CosmosDBSqlDatabaseDelete": { "$ref": "./examples/CosmosDBSqlDatabaseDelete.json" }
+        },
+        "description": "Deletes an existing Azure Cosmos DB SQL database.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseRidParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "The SQL database delete operation was successful."
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/apis/sql/databases/{databaseRid}/containers": {
+      "get": {
+        "operationId": "DatabaseAccounts_ListSqlContainers",
+        "x-ms-examples": {
+          "CosmosDBSqlContainerList": { "$ref": "./examples/CosmosDBSqlContainerList.json" }
+        },
+        "description": "Lists the SQL container under an existing Azure Cosmos DB database account.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseRidParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The SQL container properties were retrieved successfully.",
+            "schema": {
+              "$ref": "#/definitions/SqlContainerListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      },
+      "put": {
```

Comment:
As with sql databases above, PUT on a collection is a violation of resource provider contract. Resource create should be modeled as a PUT and all PUT operations must be idempotent create/update. Separation of create and update is not an option in ARM apis.

Reply:
We have PUT operation associated to 2 kind of URIs, .../databases and /databases/{database name},
which URI should we keep to represent the create and update?

Reply:
create/update should end with the resource identifier/name, i.e. /databases/{database name}